src/CensusData.cpp

```diff
@@ -725,63 +732,89 @@ void CensusData::assignTeachersAndWorkgroup (AgentContainer& pc, const int workg
 
         auto np = soa.numParticles();
 
-        auto age_group_ptr = soa.GetIntData(IntIdx::age_group).data();
-        auto workgroup_ptr = soa.GetIntData(IntIdx::workgroup).data();
-        auto work_i_ptr = soa.GetIntData(IntIdx::work_i).data();
-        auto work_j_ptr = soa.GetIntData(IntIdx::work_j).data();
-        auto school_grade_ptr = soa.GetIntData(IntIdx::school_grade).data();
-        auto school_id_ptr = soa.GetIntData(IntIdx::school_id).data();
-        auto work_nborhood_ptr = soa.GetIntData(IntIdx::work_nborhood).data();
-
-        ParallelForRNG (np, [=] AMREX_GPU_DEVICE (int ip, RandomEngine const& engine) noexcept {
-
-            int comm_to = (int)domain.index(IntVect(AMREX_D_DECL(work_i_ptr[ip], work_j_ptr[ip], 0)));
-            if (comm_to >= Ncommunity) return;
+        Gpu::HostVector<int> age_group_h(np);
+        Gpu::HostVector<int> workgroup_h(np);
+        Gpu::HostVector<int> work_i_h(np);
+        Gpu::HostVector<int> work_j_h(np);
+        Gpu::HostVector<int> school_grade_h(np);
+        Gpu::HostVector<int> school_id_h(np);
+        Gpu::HostVector<int> work_nborhood_h(np);
```

Comment:
@atmyers These allocations are inside an `AMREX_USE_OMP` pragma. If we run with multiple OMP threads, does that mean each thread will allocate arrays of size `np`?

Reply:
Yes, and there should be. `np` is the number of agents per tile, and each tile has different data for these vectors.

This whole bit of code could be optimized for the case that AMReX_USE_GPU is FALSE, though, because in that case the device to host copies below this aren't needed.